### PR TITLE
WIP Workflows refactoring

### PR DESCRIPTION
**Pull request by dset0x**
Work in progress for port of https://github.com/inveniosoftware/invenio/pull/2628 which is awaiting on https://github.com/inveniosoftware/workflow/pull/29 which is awaiting integration.

The last thing that needs to be fixed here is an issue with the database upgrade script and for some commits to be squashed and reworded.

### CODE REVIEW

File: invenio_workflows/client.py [hunk 1]
```diff
@@ -3,33 +3,25 @@
 # This file is part of Invenio.
 # Copyright (C) 2012, 2013, 2014, 2015 CERN.
 #
-# Invenio is free software; you can redistribute it and/or
+# Workflow is free software; you can redistribute it and/or
```
Comment (jirikuncar) [inline invenio_workflows/client.py]: wrong license

File: invenio_workflows/models.py [hunk 2]
```diff
@@ -52,50 +63,115 @@ class ObjectVersion(object):
...
 
-    @classmethod
-    def name_from_version(cls, version):
-        try:
-            return cls.MAPPING.keys()[cls.MAPPING.values().index(version)]
-        except ValueError:
+    @staticproperty
+    def labels():  # pylint: disable=no-method-argument
+        return {
+            0: "New",
```
Comment (jirikuncar) [inline invenio_workflows/models.py]: Wouldn't be better to write something like `INITIAL: "New",`?

Comment (dset0x) [inline invenio_workflows/models.py]: This inherits from `enum.Enum` and so I couldn't figure out a reasonable way of referencing INITIAL from within a property. The goal here is to be able to use `ObjectStatus.INITIAL.label`.

File: invenio_workflows/client.py (hunk 1 above)
Comment (dset0x) [inline invenio_workflows/client.py]: Sorry about that. It'll be fixed with the next push.

**Comment by jalavik** [comment]
Superseded by #25

**Pull request closed without merging**